aws/resource_aws_codedeploy_app.go

```diff
@@ -101,52 +110,74 @@ func resourceAwsCodeDeployAppCreate(d *schema.ResourceData, meta interface{}) er
 	// the state file. This allows us to reliably detect both when the TF
 	// config file changes and when the user deletes the app without removing
 	// it first from the TF config.
-	d.SetId(fmt.Sprintf("%s:%s", *resp.ApplicationId, application))
+	d.SetId(fmt.Sprintf("%s:%s", aws.StringValue(resp.ApplicationId), application))
 
 	return resourceAwsCodeDeployAppRead(d, meta)
 }
 
 func resourceAwsCodeDeployAppRead(d *schema.ResourceData, meta interface{}) error {
 	conn := meta.(*AWSClient).codedeployconn
+	ignoreTagsConfig := meta.(*AWSClient).IgnoreTagsConfig
 
 	application := resourceAwsCodeDeployAppParseId(d.Id())
 	log.Printf("[DEBUG] Reading CodeDeploy application %s", application)
 	resp, err := conn.GetApplication(&codedeploy.GetApplicationInput{
 		ApplicationName: aws.String(application),
 	})
 	if err != nil {
-		if codedeployerr, ok := err.(awserr.Error); ok && codedeployerr.Code() == "ApplicationDoesNotExistException" {
+		if isAWSErr(err, codedeploy.ErrCodeApplicationDoesNotExistException, "") {
 			d.SetId("")
+			log.Printf("[WARN] CodeDeploy Application (%s) not found, removing from state", d.Id())
 			return nil
-		} else {
-			log.Printf("[ERROR] Error finding CodeDeploy application: %s", err)
-			return err
 		}
+
+		log.Printf("[ERROR] Error finding CodeDeploy application: %s", err)
+		return err
+	}
+
+	app := resp.Application
+	appName := aws.StringValue(app.ApplicationName)
+
+	appArn := arn.ARN{
+		Partition: meta.(*AWSClient).partition,
+		Service:   "codedeploy",
+		Region:    meta.(*AWSClient).region,
+		AccountID: meta.(*AWSClient).accountid,
+		Resource:  fmt.Sprintf("application:%s", appName),
+	}.String()
+
+	d.Set("arn", appArn)
+	d.Set("application_id", app.ApplicationId)
+	d.Set("compute_platform", app.ComputePlatform)
+	d.Set("name", appName)
+	d.Set("github_account_name", app.GitHubAccountName)
+	d.Set("linked_to_github", app.LinkedToGitHub)
+
+	tags, err := keyvaluetags.CodedeployListTags(conn, appArn)
+
+	if err != nil {
+		return fmt.Errorf("error listing tags for CodeDeploy application (%s): %w", d.Id(), err)
 	}
 
-	d.Set("compute_platform", resp.Application.ComputePlatform)
-	d.Set("name", resp.Application.ApplicationName)
+	if err := d.Set("tags", tags.IgnoreAws().IgnoreConfig(ignoreTagsConfig).Map()); err != nil {
+		return fmt.Errorf("error setting tags: %w", err)
+	}
 
 	return nil
 }
 
 func resourceAwsCodeDeployUpdate(d *schema.ResourceData, meta interface{}) error {
 	conn := meta.(*AWSClient).codedeployconn
 
-	o, n := d.GetChange("name")
+	if d.HasChange("tags") {
+		o, n := d.GetChange("tags")
 
-	_, err := conn.UpdateApplication(&codedeploy.UpdateApplicationInput{
```

Comment:
it's interesting this was here given `name` is ForceNew...though testing this out, we could technically keep this update func, but need to remove the `ForceNew`, and update the ID during `Read` such that if `d.Id()` doesn't contain the value in `d.Get("name")`, it should be updated w/`d.SetId()`